src/main/java/htsjdk/beta/plugin/interval/HtsIntervalUtils.java

```diff
@@ -0,0 +1,132 @@
+            @Override
+            public String getContig() {
+                return interval.getQueryName();
+            }
+
+            @Override
+            public int getStart() {
+                return toIntegerSafe(interval.getStart());
+            }
+
+            @Override
+            public int getEnd() {
+                return toIntegerSafe(interval.getEnd());
+            }
+
+            @Override
+            public String toString() {
+                return String.format("%s:%s-%s",
+                        interval.getQueryName(),
+                        interval.getStart(), interval.getEnd());
+            }
+        };
+    }
+
+    /**
+     * Convert a List of {@link HtsInterval} to a list of {@link Locatable}.
+     *
+     * @param intervals list of {@link HtsInterval}s to convert
+     * @return a list of {@link Locatable}
+     */
+    @PrivateAPI
+    public static List<Locatable> toLocatableList(final List<HtsInterval> intervals) {
+        ValidationUtils.nonNull(intervals, "interval list");
+        return intervals
+                .stream()
+                .map(si -> toLocatable(si))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Convert a list of {@link HtsInterval} to an array of {@link QueryInterval}
+     *
+     * @param intervals list of {@link HtsInterval}s to convert
+     * @param dictionary sequence dictionary to use
+     * @return array of {@link QueryInterval}
+     */
+    @PrivateAPI
+    public static QueryInterval[] toQueryIntervalArray(
+            final List<HtsInterval> intervals,
+            final SAMSequenceDictionary dictionary) {
+        ValidationUtils.nonNull(intervals, "interval list");
+        ValidationUtils.nonNull(dictionary, "SAMSequenceDictionary");
+        return intervals
+                .stream()
+                .map(si -> toQueryInterval(si, dictionary))
+                .collect(Collectors.toList()).toArray(new QueryInterval[intervals.size()]);
+    }
+
+    /**
+     *
+     * @param queryIntervals list of {@link QueryInterval} to convert
+     * @param dictionary seuence dictionary to use for the conversion
+     * @return list of {@link HtsInterval}
+     */
+    @PrivateAPI
+    public static List<HtsInterval> fromQueryIntervalArray(
+            final QueryInterval[] queryIntervals,
+            final SAMSequenceDictionary dictionary) {
+        return Arrays.stream(queryIntervals)
+                .map(si -> new HtsQueryInterval(si, dictionary))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Convert a long coordinate to an integer, for use with interconverting between old style integer
+     * coordinates and new style long coordinates. Throws on overflow.
+     *
+     * @param coord long coordinate to convert
+     * @return an integer representation of {@code coord} if the conversion is safe
+     * @throws IllegalArgumentException if converting results in overflow
+     */
+    @PrivateAPI
```

Comment:
I thought of it as `@PrivateAPI` (now `@InternalAPI`) because I thought of it's usage here as a "temporary" bridge for use with underlying reader implementations that don't have native long support. But I guess thats a pipe dream. Maybe we should make it public.